selfie.c

```diff
@@ -8113,9 +8612,54 @@ void execute_symbolically() {
     do_sltu();
   } else if (is == BEQ)
     constrain_beq();
-  else if (is == JAL)
+  else if (is == JAL) {
+    pc_before_jal = pc;
     do_jal();
-  else if (is == JALR) {
+
+    pc_after_jal = pc;
+
+    // we need to check if we jump into a recursion
+    fetch();
+    decode();
+
+    if (is == ADDI) {
```

Comment:
You can simplify this by just checking if the JAL uses the RA or the ZERO register. In the former case it is a procedure call.

Reply:
Just search for `emit_JAL` to see that.

Reply:
That is much better! I did not think of that, I changed it.